test_integration/02_stake_and_mint/01_stake_and_mint.js

```diff
@@ -0,0 +1,203 @@
+//
+// ----------------------------------------------------------------------------
+
+const BN = require('bn.js');
+const shared = require('../shared');
+const Utils = require('../../test/test_lib/utils');
+const EventDecoder = require('../../test/test_lib/event_decoder');
+
+const AssertStake = require('./utils/assert_stake');
+const AssertProveGateway = require('./utils/assert_prove_gateway');
+const AssertConfirmStakeIntent = require('./utils/assert_confirm_stake_intent');
+const AssertProgressStake = require('./utils/assert_progress_stake');
+const AssertProgressMint = require('./utils/assert_progress_mint');
+const ProofUtils = require('./utils/proof_utils');
+const Anchor = require('./utils/anchor');
+
+describe('Stake and mint', async () => {
+    let assertStake;
+    let assertProgressStake;
+    let assertProgressMint;
+    let proofUtils;
+
+    let originAccounts;
+    let auxiliaryAccounts;
+    let originWeb3;
+    let auxiliaryWeb3;
+
+    let gateway;
+    let cogateway;
+    let token;
+    let baseToken;
+    let auxiliaryAnchor;
+    let ostPrime;
+    let stakeRequest;
+
+    before(async () => {
+        originWeb3 = shared.origin.web3;
+        auxiliaryWeb3 = shared.auxiliary.web3;
+        originAccounts = shared.origin.accounts;
+        auxiliaryAccounts = shared.origin.accounts;
+        token = shared.origin.contracts.Token;
+        baseToken = shared.origin.contracts.BaseToken;
+        gateway = shared.origin.contracts.EIP20Gateway;
+        cogateway = shared.auxiliary.contracts.EIP20CoGateway;
+        ostPrime = shared.auxiliary.contracts.OSTPrime;
+
+        const hasher = Utils.generateHashLock();
+        stakeRequest = {
+            amount: new BN(200),
+            gasPrice: new BN(1),
+            gasLimit: new BN(100),
+            staker: originAccounts[0],
+            bounty: await gateway.bounty.call(),
+            nonce: await gateway.getNonce.call(originAccounts[0]),
+            beneficiary: auxiliaryAccounts[2],
+            hashLock: hasher.l,
+            unlockSecret: hasher.s,
+        };
+
+        assertStake = new AssertStake(gateway, token, baseToken);
+        assertProgressStake = new AssertProgressStake(gateway, token, baseToken);
+        assertProgressMint = new AssertProgressMint(auxiliaryWeb3, cogateway, ostPrime);
+        proofUtils = new ProofUtils(originWeb3, auxiliaryWeb3);
+        auxiliaryAnchor = new Anchor(
+            originWeb3,
+            shared.auxiliary.contracts.Anchor,
+        );
+    });
+
+    it('stake', async () => {
+        // Capture initial token and base token balance of staker and gateway.
+        const initialBalances = await assertStake.captureBalances(stakeRequest.staker);
+        // Approve gateway for stake amount.
+        await token.approve(
+            gateway.address,
+            stakeRequest.amount,
+            { from: stakeRequest.staker },
+        );
+
+        // Approve gateway for bounty.
+        await baseToken.approve(
```

Comment:
When you have to add a single line comment above a piece of code, it often makes sense to extract the code into a function with a meaningful name instead. For example, you could extract the `baseToken.approve` into a function called `approveGatewayForBounty()`.